DEVELOPER.md

```diff
@@ -89,21 +89,63 @@ You can use a Vagrant box running Kong and Postgres that you can find at
 #### Source Install
 
 Kong is mostly an OpenResty application made of Lua source files, but also
-requires some additional third-party dependencies. We recommend installing
-those by following the [source install instructions](https://docs.konghq.com/install/source/).
+requires some additional third-party dependencies, some of which compiled
+with tweaked options, and runs on a modified version of OpenResty with patches.
 
-Instead of following the second step (Install Kong), clone this repository
-and install the latest Lua sources instead of the currently released ones:
+You can build OpenResty, OpenSSL, PCRE and Luarocks with [Kong/kong-ngx-build](https://github.com/Kong/kong-build-tools/tree/master/openresty-build-tools), which we will utilize for the following process.
+
+Below is a template for how can install from the source.
 
 ```shell
+# Clone this repository and Kong/kong-ngx-build.
+git clone https://github.com/Kong/kong-build-tools.git
 git clone https://github.com/Kong/kong
-cd kong/
 
-# you might want to switch to the development branch. See CONTRIBUTING.md
+cd kong
+# You might want to switch to the development branch. See CONTRIBUTING.md
 git checkout master
 
-# install the Lua sources
-luarocks make
+# To build dependencies we need to inspect sources versions that is requireed from
+# `.requirements`, and use that as arguments to call a build sciprt. 
+# You can mannually do this, or follow the steps below.
+
+# Somewhere you're able or prefer to build
+BUILDROOT=$(realpath ~/kong-dep)
+mkdir ${BUILDROOT}
+
+RESTY_VERSION=$(grep -oP 'RESTY_VERSION=\K.*' .requirements)
+RESTY_OPENSSL_VERSION=$(grep -oP 'RESTY_OPENSSL_VERSION=\K.*' .requirements)
+RESTY_LUAROCKS_VERSION=$(grep -oP 'RESTY_LUAROCKS_VERSION=\K.*' .requirements)
+RESTY_PCRE_VERSION=$(grep -oP 'RESTY_PCRE_VERSION=\K.*' .requirements)
+
+cd ../kong-build-tools/openresty-build-tools
+
+# Before we run the script, make sure curl and unzip is installed.
+# Also, to build we need gcc/g++ and m4 installed.
+# Here is an example for ubuntu:
+sudo apt install curl unzip g++ m4
+
+# You might want to add also --debug
+./kong-ngx-build -p ${BUILDROOT} --openresty ${RESTY_VERSION} --openssl ${RESTY_OPENSSL_VERSION} --luarocks ${RESTY_LUAROCKS_VERSION} --pcre ${RESTY_PCRE_VERSION}
```

Comment:
Duplicated with section ["Dependencies (Build from source)"](https://github.com/Kong/kong/blob/master/DEVELOPER.md#dependencies-binary-release)

Reply:
> 

I tried to preserve the original structure of the doc and referred to this section in the source build.